tests/test_cvc.ml

```diff
@@ -0,0 +1,109 @@
+open OUnit2
+
+let rsa_suite =
+  let open Key_parsers in
+  let open Asn1.RSA_CVC in
+  let expected_public =
+    let n =
+      Z.of_string
+        "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\
+        FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\
+        FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\
+        FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+    in
+    let e = Z.of_string "0x010001" in
+    Public.{n; e}
+  in
+  let cmp = Z.equal in
+  let printer = Z.to_string in
+  let test_pub ~decode (expected : Public.t) cvc ctxt =
+    let real = decode cvc in
+    let open Public in
+    Test_util.assert_ok real @@ function
+      | `RSA real ->
+          assert_equal ~ctxt ~cmp ~printer ~msg:"n" expected.n real.n;
+          assert_equal ~ctxt ~cmp ~printer ~msg:"e" expected.e real.e
+      | `ECDSA _
+      | `UNKNOWN ->
+          assert_failure ""
+  in
+  let cvc_suite =
+    let cvc = Cstruct.of_string @@ [%blob "../tests/keys/rsa_cvc_dummy.key"] in
+    [ "Public" >:: test_pub ~decode:Asn1.CVC.decode expected_public cvc
+    ]
+  in
+  [ "CVC" >::: cvc_suite
+  ]
+
+let ecdsa_suite =
+  let open Key_parsers in
+  let open Asn1.ECDSA_CVC in
+  let expected_public =
+    let modulus =
+      Z.of_string
+        "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+    in
+    let coefficient_a =
```

Comment:
Here you can use actual EC parameters, for example the `secp256r1` ones:
- `modulus = coefficient_a = 0xffffffff00000001000000000000000000000000fffffffffffffffffffffffc`
- `coefficient_b = 0x5ac635d8aa3a93e7b3ebbd55769886bc651d06b0cc53b0f63bce3c3e27d2604b`
- `base_point_g = 0x46b17d1f2e12c4247f8bce6e563a440f277037d812deb33a0f4a13945d898c2964fe342e2fe1a7f9b8ee7eb4a7c0f9e162bce33576b315ececbb6406837bf51f5`
- `base_point_r_order = 0xffffffff00000000ffffffffffffffffbce6faada7179e84f3b9cac2fc632551`
- `cofactor_f = 1`

And you can use a random value for `cofactor_f` for the same reasons as RSA keys.